src/display.rs

```diff
@@ -1,35 +1,51 @@
 use std::{
     collections::{HashMap, HashSet},
     fmt::Display,
-    ops::Deref,
+    iter::FromIterator as _,
+    ops::Deref as _,
+    str::FromStr as _,
 };
 
 use crate::utils::add_extra_where_clauses;
 use proc_macro2::{Ident, Span, TokenStream};
 use quote::{quote, quote_spanned};
 use syn::{
-    parse::{Error, Result},
-    spanned::Spanned,
-    Attribute, Data, DeriveInput, Fields, GenericArgument, Lit, Meta, MetaNameValue, NestedMeta, Path, PathArguments, Type, TypeReference
+    parse::{
+        Error,
+        Result,
+        Parser as _,
+    },
+    punctuated::Punctuated,
+    Attribute,
+    Data,
+    DeriveInput,
+    Fields,
+    GenericArgument,
+    GenericParam,
+    Lit,
+    Meta,
+    MetaList,
+    MetaNameValue,
+    NestedMeta,
+    Path,
+    PathArguments,
+    PathSegment,
+    Token,
+    TraitBound,
+    TraitBoundModifier,
+    Type,
+    TypeParamBound,
+    TypePath,
```

Comment:
Why import all this? Is using `syn::Type` or `syn::TypeParamBound` is too bad and requires `syn::` part to be removed?

This has sense if there is a really long paths, to shorten it for better readability. The `punctuated::Punctuated` is a good example. However, in much cases `syn::Something` is quite good enough.

Reply:
Just followed existing style here. I'm for explicitly prefixing `syn::` personally.

Reply:
@JelteF wouldn'y you mind on some cosmetic refactoring for `src/display.rs` file?

Reply:
Sounds good for the changes made here, but if it touches code in another part of the file let's do that in another PR.

Reply:
@ffuugoo ☝️